File: ToolKit/src/main/java/com/uday/android/toolkit/listeners/ConfirmListener.java
package com.uday.android.toolkit.listeners;

import android.content.*;
import android.util.*;
import android.view.inputmethod.*;
import com.uday.android.toolkit.*;
import com.uday.android.toolkit.fragments.*;
import com.uday.android.toolkit.ui.*;
import android.app.*;
import android.widget.*;

public class ConfirmListener implements DialogUtils.OnClickListener
{
	private AndroidImagesFragment fragment;
	public ConfirmListener(AndroidImagesFragment fragment){
		this.fragment=fragment;
	}
	
	@Override
	public void onClick(final AlertDialog utils){
		utils.cancel();
		switch(fragment.mOption){
			case fragment.SELECTED_REPACK:
				Object[] obj=DialogUtils.showTermDialog(fragment.context,"Repack img","Repacking project please wait...","open folder","finish");
				fragment.termDialog=(AlertDialog)obj[0];
				fragment.termTextView=(TextView)obj[1];
				fragment.termProgress=(ProgressBar)obj[2];
				new Thread(fragment.executer).start();
				break;
			case fragment.SELECTED_INSTALL:
				fragment.pDialog=new ProgressDialog(fragment.context).show(fragment.context,"Installing","Installing selected "+fragment.choosen+"\nplease wait...",false,false);
				new Thread(fragment.executer).start();
				break;
			case fragment.SELECTED_BACKUP:
				fragment.pDialog=new ProgressDialog(fragment.context).show(fragment.context,"Backup","Backup process in progress please wait...",false,false);
				new Thread(fragment.executer).start();
				break;
			case fragment.SELECTED_RESTORE:
				fragment.pDialog=new ProgressDialog(fragment.context).show(fragment.context,"Restoring","Restoring "+fragment.BLOCK_NAME.toUpperCase()+".\nplease wait...",false,false);
				new Thread(fragment.executer).start();
				break;
			case fragment.SELECTED_RESTORE_ITEM:
				fragment.pDialog=new ProgressDialog(fragment.context).show(fragment.context,"Restoring","Restoring "+fragment.BLOCK_NAME.toUpperCase()+".\nplease wait...",false,false);
				new Thread(fragment.executer).start();
		}
	}
}

File: app/src/main/java/com/uday/android/toolkit/listeners/ConfirmListener.kt
package com.uday.android.toolkit.listeners

import android.annotation.SuppressLint
import android.widget.ProgressBar
import android.widget.TextView
import androidx.appcompat.app.AlertDialog
import com.uday.android.toolkit.fragments.AndroidImagesFragment
import com.uday.android.toolkit.ui.DialogUtils

class ConfirmListener(private val fragment: AndroidImagesFragment) : DialogUtils.OnClickListener {

    @SuppressLint("DefaultLocale")
    override fun onClick(p1: AlertDialog?) {
        p1?.cancel()
        when (fragment.mOption) {
            AndroidImagesFragment.SELECTED_REPACK -> {
                val obj = DialogUtils.showTermDialog(fragment.context, "Repack img", "Repacking project please wait...", "open folder", "finish")
                fragment.termDialog = obj[0] as AlertDialog
                fragment.termTextView = obj[1] as TextView
                fragment.termProgress = obj[2] as ProgressBar
                Thread(fragment.executor).start()
            }
            AndroidImagesFragment.SELECTED_INSTALL -> {
                @Suppress("DEPRECATION")
                fragment.pDialog = android.app.ProgressDialog.show(fragment.context, "Installing", "Installing selected " + fragment.choosen + "\nplease wait...", false, false)
                Thread(fragment.executor).start()
            }
            AndroidImagesFragment.SELECTED_BACKUP -> {
                @Suppress("DEPRECATION")
                fragment.pDialog = android.app.ProgressDialog.show(fragment.context, "Backup", "Backup process in progress please wait...", false, false)
                Thread(fragment.executor).start()
            }
            AndroidImagesFragment.SELECTED_RESTORE -> {
                @Suppress("DEPRECATION")
                fragment.pDialog = android.app.ProgressDialog.show(fragment.context, "Restoring", "Restoring " + fragment.blockName!!.toUpperCase() + ".\nplease wait...", false, false)
                Thread(fragment.executor).start()
            }
            AndroidImagesFragment.SELECTED_RESTORE_ITEM -> {
                @Suppress("DEPRECATION")
                fragment.pDialog = android.app.ProgressDialog.show(fragment.context, "Restoring", "Restoring " + fragment.blockName!!.toUpperCase() + ".\nplease wait...", false, false)
                Thread(fragment.executor).start()
            }
        }
    }
}
